sqlx-sqlite/src/connection/mod.rs

```diff
@@ -60,6 +67,19 @@ pub(crate) struct ConnectionState {
     pub(crate) statements: Statements,
 
     log_settings: LogSettings,
+
+    /// Stores the progress handler set on the current connection. If the handler returns `false`,
+    /// the query is interrupted.
+    progress_handler_callback: Option<Handler>,
+}
+
+impl ConnectionState {
+    /// Drops the `progress_handler_callback` if it exists.
+    pub(crate) fn drop_progress_handler_callback(&mut self) {
```

Comment:
I'd rename this to `remove_progress_handler` and have it also remove the progress handler from the database connection as mentioned in a previous comment.

Reply:
Can do that, was under the assumption that the connection is being dropped/destroyed anyway.